src/Stratis.Bitcoin/Utilities/IpHelper.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System.Linq;
+using System.Net;
+using System.Net.Sockets;
+using NBitcoin;
+
+namespace Stratis.Bitcoin.Utilities
+{
+    public static class IpHelper
+    {
+        /// <summary>
+        /// Find ports that are free to use.
+        /// </summary>
+        /// <param name="ports">A list of ports to checked or fill/replace as necessary.</param>
```

Comment:
Not clear comment, is this an existing list of ports?

Reply:
You pass this method a list of ports, and this method will either:
- replace the ports that are used by new, unused ones
- add new, unused ports to the list

Reply:
ok yeah, I was just referring to the comment not being clear on the behaviour.

Reply:
I simply took the method and comment from TestsHelper, but yeah a beefier comment maybe